internal/infra/push/datadestination_sql.go

```diff
@@ -327,6 +327,7 @@ func (rw *SQLRowWriter) Write(row push.Row) *push.Error {
 
 	_, err2 := rw.statement.Exec(values...)
 	if err2 != nil {
+		rw.statement = nil // reset statement after error
```

Comment:
You should always call Close() method on a statement and by setting nil here, it will never be closed. Is this a problem ?

Reply:
You're right ! And the problem is more complex ...

- Commit on error ?
- Rollback to previous commit and replay uncommitted lines without the last one ?
- Duplicated is a special case  : one entity could contains duplicate lines from different path.

That behavior should be manage by driver not by infra.